### ISSUE TITLE
What does this extension actually do internally?

### ISSUE THREAD

**Issue by nextgenthemes** [Devine-Davies/vs-make-hidden#19]
Is creates its own config file. Where exactly? In the config folder probably, I will find it after this sire .... And why does it create its own config file when vscode has its own that work globally and for the current workspace?

Is appears it does not use a local config file for the current workspace? Why?

It appears it does not use vscode internal settings for `exclude-files` or `files-excludes` whatever its called. Why?

What are workspaces in relation to this extension? Does it create its down workspaces? I do not get it, creating a workspace I can't see a new file creating it again just give an error that the config cant be created.

Its incredibly confusing because nothing of this is explained.

**Comment by nextgenthemes** [comment]
Actually it seems to fail to create the file in `.vscode` as the message says. The issue may have to do with that fact that I use VScodium that uses the `~/.vscode-oss` dir and I see all the extensions in there. 

I guess this should be redirected internally, but it does not, creating `~/.vscode` makes it still fail. I should probably open an issue with them.

... Actually is DOES save the settings after I exit and restart it.

ROFL this gets more confusing. I actually just saw it does actually do almost what I expected it used the local `.vscode/settings.json` to set the excludes. So vscode has 3 different ways to set settings? I only knew about in workspace files and globally. I would prefer this to set it in the workspace files. Still no clue what how Workspaces creation relates to this extension and what the file it's supposed to store is for, but I am glad it uses the native files excludes. 

It seems to work, despite the error, but Workspaces do not work `MakeHidden.json` can not be created.

**Comment by alexkuc** [comment]
> So vscode has 3 different ways to set settings? I only knew about in workspace files and globally.

Yes, this appears to be a known confusion. You might want to check out https://github.com/microsoft/vscode/issues/113202 where vscode contributors addressed the issue.

> Is creates its own config file. Where exactly?

There is a dedicated function which determines which `settings.json` file to use:

https://github.com/Devine-Davies/vs-make-hidden/blob/9adab08b0cd4fa6f119342b7367241c9ec71fd71/src/MakeHidden/utilities.ts#L82-L92

> What are workspaces in relation to this extension? Does it create its down workspaces? I do not get it, creating a workspace I can't see a new file creating it again just give an error that the config cant be created.

I think this relates more to vscode manual. They have a section called [Users and Workspace Settings](https://vscode.readthedocs.io/en/latest/getstarted/settings/) where behavior of workspaces is explained.

> The issue may have to do with that fact that I use VScodium that uses the ~/.vscode-oss dir and I see all the extensions in there.

This sounds like issue with vscodium specifically and not vscode as I didn't experience any writing issues. Try this extension with vscode and see if it works for you.
